src/azul/indexer/log_forwarding_service.py

```diff
@@ -0,0 +1,98 @@
+import csv
+import gzip
+from typing import (
+    Iterator,
+)
+
+from botocore.response import (
+    StreamingBody,
+)
+from furl import (
+    furl,
+)
+
+from azul.deployment import (
+    aws,
+)
+from azul.types import (
+    MutableJSON,
+)
+
+
+class LogForwardingService:
+    """
+    Parse logs from an Application Load Balancer.
+    Adapted from https://github.com/rupertbg/aws-load-balancer-logs-to-cloudwatch
+    """
+
+    @property
+    def s3(self):
+        return aws.s3
+
+    def read_logs(self, bucket: str, key: str) -> Iterator[MutableJSON]:
+        response = self.s3.get_object(Bucket=bucket, Key=key)
+        body: StreamingBody = response['Body']
+        with gzip.open(body, mode='rt') as f:
+            for row in csv.reader(f, delimiter=' ', quotechar='"'):
```

Comment:
I take it that the quotes are used to escape spaces in ALB logs. Is there documentation on how literal quotes are escaped? IIRC, quotes in CSV are escaped by doubling them up.

Reply:
I can't find any documentation on how quotes are escaped by the ALB. You're correct about how CSV handles it.

Reply:
One way or another, we need to determine how quotes are handled. If we can't find documentation, we need to run an experiment.